components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticationFrameworkWrapper.java

```diff
@@ -156,4 +158,29 @@ public String getQueryString() {
     public void addHeader(String key, String values) {
         modifiableHeaders.put(key, values);
     }
+
+    /**
+     * Modified cookies from the request wrapper. Exclude and return the commonAuthId cookie,
+     * if the request attribute "removeCommonAuthCookie" is set to true.
+     */
+    @Override
+    public Cookie[] getCookies() {
+
+        // Exclude and return the commonAuthId cookie, if the request attribute "removeCommonAuthCookie" is set to true.
+        Cookie[] cookies = super.getCookies();
+
+        if (cookies == null || !Boolean.TRUE.toString().equals(String.valueOf(
```

Comment:
Might be able to use
Boolean.TRUE.equals(request.getAttribute(FrameworkConstants.REMOVE_COMMONAUTH_COOKIE))